src/character.cpp

```diff
@@ -9684,6 +9684,44 @@ bool Character::use_charges_if_avail( const itype_id &it, int quantity )
     return false;
 }
 
+std::pair<float, item> Character::get_best_weapon_by_damage_type( const damage_type_id dmg_type )
+const
+{
+    std::pair<float, item> best_weapon = std::make_pair( 0, item() );
+
+    cache_visit_items_with( "best_damage", &item::is_melee, [&best_weapon, this,
```

Comment:
key needs damage type in it.

Reply:
i am still not very sure what key does here